pkg/kubectl/cmd/delete.go

```diff
@@ -123,17 +124,6 @@ func RunDelete(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 	}
 
 	ignoreNotFound := cmdutil.GetFlagBool(cmd, "ignore-not-found")
-	if deleteAll {
```

Comment:
Note that that is only when doing a `delete pods --all`, etc. it has to list to determine the names to delete. When given an explicit list (e.g. `delete -f list.yaml`), it shouldn't be making any API calls other than the delete calls